### PR TITLE
refactor(core/projects): Convert projects header and configuration to react

### PR DESCRIPTION
**Pull request by christopherthielen**
This converts projects header, project configuration modal, and routing to react.
The project dashboard still remains as angularjs.

The PR It depends on #5882 and  #5883.  There are still some CSS issues I need to take care of, but this PR can be reviewed without the CSS fixes.

I left most of the commits separate in case anyone wants to see the individual steps instead of a big bang PR.  Will squash during merge, obviously.

This was done in collaboration with @archana-s

### CODE REVIEW

**Comment by christopherthielen** [comment]
hmm, seems to be missing a couple commits.  will check on this tomorrow.

File: app/scripts/modules/core/src/projects/configure/ProjectAttributes.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  public validate(values: IProject) {
+    const errors: FormikErrors<IProject> = {};
+
+    const isValidName = (name: string) => {
+      const namePattern = /^[^\\\^/^?^%^#]*$/;
+      return name.match(namePattern);
+    };
+
+    const isValidEmail = (email: string) => {
+      const emailPattern = /^(.+)\@(.+).([A-Za-z]{2,6})/;
```
Comment (jrsquared) [inline app/scripts/modules/core/src/projects/configure/ProjectAttributes.tsx]: Not in this PR, but we really need to pull these types of validators into a utility.

File: yarn.lock [hunk 2]
```diff
@@ -5,6 +5,7 @@
 "3d-view-controls@^2.2.0":
   version "2.2.2"
   resolved "https://registry.yarnpkg.com/3d-view-controls/-/3d-view-controls-2.2.2.tgz#7173fc197e9e7e4dbc63213438467047dbc84fa2"
+  integrity sha512-WL0u3PN41lEx/4qvKqV6bJlweUYoW18FXMshW/qHb41AVdZxDReLoJNGYsI7x6jf9bYelEF62BJPQmO7yEnG2w==
```
Comment (jrsquared) [inline yarn.lock]: oof. i'm guessing you updated your yarn version to get all these `integrity` properties? Any chance you could split that bit into a separate PR?

File: app/scripts/modules/core/src/presentation/forms/inputs/ReactSelectInput.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    };
+    return (field.onChange || noop)(fakeEvent);
+  };
+};
+
+/**
+ * A react-select Input
+ *
+ * This input supports error validation state rendering
+ * It also has a simplified stringOptions prop for when Option interface is overkill.
```
[the remark this answers is not in the document]
Comment (christopherthielen) [reply app/scripts/modules/core/src/presentation/forms/inputs/ReactSelectInput.tsx]: it's a lie though.  It doesn't have a `stringOptions` prop :) . Will fix in upcoming PR

File: app/scripts/modules/core/src/presentation/forms/inputs/ReactSelectInput.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ */
+export class ReactSelectInput extends React.Component<IReactSelectInputProps> {
+  public static defaultProps: Partial<IReactSelectInputProps> = {
+    tethered: true,
+  };
+
+  public render() {
+    const { tethered, field, validation, inputClassName, ...otherProps } = this.props;
+
+    // TODO: see if implementing onBlur makes sense
```
[the remark this answers is not in the document]
Comment (christopherthielen) [reply app/scripts/modules/core/src/presentation/forms/inputs/ReactSelectInput.tsx]: implemented, will file a new PR

File: app/scripts/modules/core/src/presentation/forms/inputs/StringsAsOptions.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Converts an array of strings to an array of Options[] suitable for use with react-select
+ *
+ * Example:
+ * <StringsAsOptions strings={apps}>
+ *   {options => <SelectInput onChange={...} clearable={false} options={options}/>}
+ * </StringsAsOptions>
+ */
+export class StringsAsOptions extends React.Component<IStringOptionsProps, IStringOptionsState> {
+  private memoizedMakeOptions = memoizeOne(makeOptions, isEqual);
```
[the remark this answers is not in the document]
Comment (christopherthielen) [reply app/scripts/modules/core/src/presentation/forms/inputs/StringsAsOptions.tsx]: It's like lodash `memoize`, but only remembers a single value (the last one).  This is an alternate to updating `this.state` when `this.props` changes.

https://github.com/alexreardon/memoize-one taken from this blog post: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

File: app/scripts/modules/core/src/projects/configure/ConfigureProjectModal.css [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  color: var(--color-accent);
+}
+
+.Pipelines table thead td,
+.Clusters table thead td {
+  border-bottom: 1px solid var(--color-divider);
+}
+
+.Pipelines .button.zombie,
+.Clusters .button.zombie {
```
[the remark this answers is not in the document]
Comment (christopherthielen) [reply app/scripts/modules/core/src/projects/configure/ConfigureProjectModal.css]: Agreed.  Will look into refactoring the CSS
